ddclient.in

```diff
@@ -2273,65 +2273,79 @@ sub get_ip {
     return $ip;
 }
 
+
+######################################################################
+## Regex to find IPv4 address. Accepts embedded leading zeros.
+######################################################################
+my $regex_ipv4 = qr/(?:(?<octet>25[0-5]|2[0-4]\d|[01]?\d\d?)\.){3}(?&octet)/;
+
+######################################################################
+## Regex to find IPv6 in standard, compressed format. Accepts
+## embedded leading zeros in the address. Adapted from regex in the
+## Regular Expressions Cookbook by J.Goyvaerets and S.Levithan
+######################################################################
+my $regex_ipv6_only = qr/(?<![:.0-9A-Z])        ## Negative lookbehind for word boundry
```

Comment:
I think everything after the negative lookbehind here and before the negative lookahead at the end must be inside a group: `(?:...)`.